scripts/SANS/sans/algorithm_detail/single_execution.py

```diff
@@ -375,27 +356,33 @@ def run_optimized_for_can(reduction_alg, reduction_setting_bundle, event_slice_o
     # |  True        |        False                        |           True                      |    True     |
     # |  True        |        False                        |           False                     |    False    |
 
-    output_parts_bundle = reduced_slices[0].parts_bundle
-    is_invalid_partial_workspaces = ((output_parts_bundle.output_workspace_count is None
-                                      and output_parts_bundle.output_workspace_norm is not None)
-                                     or (output_parts_bundle.output_workspace_count is not None
-                                         and output_parts_bundle.output_workspace_norm is None))
-    output_transmission_bundle = reduced_slices[0].transmission_bundle
-    is_invalid_transmission_workspaces = (output_transmission_bundle.calculated_transmission_workspace is None
-                                          or output_transmission_bundle.unfitted_transmission_workspace is None)
-    partial_output_require_reload = output_parts and is_invalid_partial_workspaces
-
-    must_reload = reduced_slices[0].output_bundle.output_workspace is None \
-                  or partial_output_require_reload or is_invalid_transmission_workspaces
+    def needs_reload(slice: ReducedSlice):
+        output_parts_bundle = slice.parts_bundle
+        is_invalid_partial_workspaces = ((output_parts_bundle.output_workspace_count is None
+                                          and output_parts_bundle.output_workspace_norm is not None)
+                                         or (output_parts_bundle.output_workspace_count is not None
+                                             and output_parts_bundle.output_workspace_norm is None))
```

Comment:
I wonder if this might be made a bit less verbose with the xor operator, something like 
```is_invalid_partial_workspaces = bool(output_parts_bundle.output_workspace_count) ^ bool(output_parts_bundle.output_workspace_norm)```
though whether this conforms to accepted python style/readability I've no idea. Let me know what you think!

Reply:
In this case because it's internal I'm happy to let a developer see an XOR and it's much cleaner than this existing approach
Good spot, thanks :+1: